src/userApps.ts

```diff
@@ -73,8 +73,24 @@ export class UserApps {
         await this.storage.appRecords;
     }
 
-    static async onPressLogin({ redirect = true, callbackPath }: OnPressLoginOptions): Promise<string | void> {
+    static async onPressLogin(
+        { redirect = true, callbackPath }: OnPressLoginOptions,
+        keyManager?: KeyManager
+    ): Promise<string | void> {
+        //TODO: dont create new key if it exist
         const { privateKey, publicKey } = generateRandomKeyPair();
+
+        if (keyManager) {
+            await keyManager.storeKey({
+                level: KeyManagerLevel.LOCALSTORAGE,
+                privateKey: privateKey,
+            });
+        } else {
+            if (!localStorage.getItem(KeyManagerLevel.LOCALSTORAGE)) {
```

Comment:
this should all be done through the keyManager interface
L83-93 should be
```ts
 await keyManager.storeKey({
                level: KeyManagerLevel.LOCALSTORAGE,
                privateKey: privateKey,
```
this should handle the case where there is already something local storage, in this case I would expect that the key in local storage is ignored and replaced with `storeKey()`